### PR TITLE
OpTracker: rename osd_num_op_tracker_shard to num_op_tracker_shard

### PR DESCRIPTION
**Pull request by Rethan**
since it also applies to mds.

Both mds and osd use osd_num_op_tracker_shard as op_tracker shards, so rename it as num_op_tracker_shard to remove the osd prefix.

Signed-off-by: haoyixing <haoyixing@kuaishou.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by dparmar18 — COMMENTED** [review]
Sharding breaks down data into multiple locations in order to distribute the load and also increases parallelism. It's just an opinion of mine that as it is mostly associated with OSDs, I feel like `osd_num_op_tracker_shard` sounds more meaningful than just `num_op_tracker_shard`.

**Comment by vshankar** [comment]
Maybe. @neha-ojha wdyt?

**Review by nmshelke — APPROVED** [review]
Looks good to me if it only related to rename existing term and make it more generic.

**Comment by Rethan** [comment]
jenkins test make check

**Comment by lxbsz** [comment]
Will this break something ? Such as the third part apps ?
Maybe it makes sense to doc it in `PendingReleaseNotes`.

**Comment by Rethan** [comment]
> Will this break something ? Such as the third part apps ? Maybe it makes sense to doc it in `PendingReleaseNotes`.

Updated.

File: src/common/options/global.yaml.in [hunk 1]
```diff
@@ -3228,7 +3228,7 @@ options:
   default: true
   with_legacy: true
 # The number of shards for holding the ops
-- name: osd_num_op_tracker_shard
+- name: num_op_tracker_shard
```
Comment (rzarzynski) [inline src/common/options/global.yaml.in:3231]: What about already existing clusters? Will presence of the old option rise a warning, an error or it's just going to be silently ignored?

Comment (Rethan) [reply src/common/options/global.yaml.in:3231]: Silently ignored, from my point of view.

**Pull request closed without merging**